dapps/marketplace/src/pages/nav/Nav.js

```diff
@@ -66,12 +67,67 @@ const Nav = ({
   history
 }) => {
   const [open, setOpen] = useState()
+  const [consoleOpen, setConsoleOpen] = useState(false)
+  const [consoleLogConnected, setConsoleLogConnected] = useState(false)
+
+  const screenConsoleEnabled = localStorage.screenConsole
+
+  useEffect(() => {
+    if (wallet && !consoleLogConnected && screenConsoleEnabled) {
+      ConsoleLogCatcher().connect((method, logString) => {
+        const logs = JSON.parse(localStorage.getItem('capturedLogs') || '[]')
+        // only keep max 15 items in the logs
+        logs.slice(Math.max(0, logs.length - 15))
+        logs.push({ method, log: logString })
+        localStorage.setItem('capturedLogs', JSON.stringify(logs))
```

Comment:
This is rewriting the entire log in localstorage for every single line logged. I have no idea about the performance of localStorage on mobile... but if this turns out to be super slow, here is a possible option to make this a less io intensive: get the date in msec, modulo it 15 and use that as part of the local storage key. Pseudo code:
```
timestamp = Date.now()
key = 'capturedLogs' + timestamp  % 15
val = JSON.stringify( { timestamp, method, log: logString } )
localStorage.setItem(key, val)
```

Then at display time, read all the logs and sort by timestamp.
```
logs = []
for (i=0; i< 15; i++) {
   key = 'capturedLogs' + i
   logs[i] = localStorage.getItem(key)
}
logs = logs.filter(l => l).sort((l1, l2) => l1.timestamp > l2.timestamp)
```

Reply:
That is a pretty cool optimisation @franckc. 

Did a quick performance test on my (not so flagship) iPhone6: 
```
const log = { method:'info', log: 'The purpose of this is to really test how fast various approaches to storing stuff in local storage are. This should be enough text for performance testing' }

//prepare 15 logs in storage
const largeLogEntry = []
for (let i = 0; i < 15; i++) {
  largeLogEntry.push(log)
}
localStorage.setItem(`singleCapturedLogsTest`, JSON.stringify(largeLogEntry))


const singleLogTest = () => {
  for (let i = 0; i < 15; i++) {
    localStorage.setItem(`largeCapturedLogsTest`, JSON.stringify(largeLogEntry))
  }
}

const moduloTest = () => {
  // writing only 1 log
  for (let i = 0; i < 15; i++) {
    localStorage.setItem('smallCapturedLogsTest', JSON.stringify(log))
  }
}

var t0 = Date.now();
singleLogTest();
var t1 = Date.now();
console.log("Single test took " + (t1 - t0) + " milliseconds.");

t0 = Date.now();
moduloTest();
t1 = Date.now();
console.log("Modulo test took " + (t1 - t0) + " milliseconds.");
```

Your method really works ~5times faster. The differences though for storing 15 logs was 0.2ms with optimised method and 1ms for non optimised one. I am leaning a bit towards not introducing more code complexity...

Will merge this for now. If you feel this should still be optimised let me know and I will optimise it according to your suggestion.

Reply:
Thanks for benchmarking ! Yes totally agreed it is not worth the extra complexity given your benchmark result. Good call.